### PR TITLE
Rewrite `validate-search-params.js` in TypeScript

### PR DESCRIPTION
**Pull request by Cryrivers**
#### Checklist

- [X] I have read the documentation.
- [X] I have included a pull request description of my changes.
- [ ] I have included some tests.
- [ ] If it's a new feature, I have included documentation updates.

### CODE REVIEW

File: source/utils/validate-search-params.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-const is = require('@sindresorhus/is');
+import is from '@sindresorhus/is';
 
-const verify = (value, type) => {
-	if (!is.string(value) && !is.number(value) && !is.boolean(value) && !is.null(value)) {
+type SearchParams = {
+	[key: string]: string | number | boolean | null
+}
+
+const verify = (value: any, type: string) => {
```
Comment (sindresorhus) [inline source/utils/validate-search-params.ts]: Use `unknown` instead of `any`.

Comment (Cryrivers) [reply source/utils/validate-search-params.ts]: Fixed

**Comment by sindresorhus** [comment]
There are also some lint errors: https://travis-ci.org/sindresorhus/got/jobs/482315362

Just run `npx xo --fix` locally to fix them.

**Comment by Cryrivers** [comment]
> There are also some lint errors: https://travis-ci.org/sindresorhus/got/jobs/482315362
> 
> Just run `npx xo --fix` locally to fix them.

Sorry, my oversight. It has been fixed.

**Comment by sindresorhus** [comment]
Nice work! Thanks for the contribution :)

**Pull request merged**